lib/upload.ts

```diff
@@ -985,6 +1014,133 @@ export class BaseUpload {
   _sendRequest(req: HttpRequest, body?: SliceType): Promise<HttpResponse> {
     return sendRequest(req, body, this.options)
   }
+
+  /**
+   * Apply default stall detection options
+   */
+  private _getStallDetectionDefaults(
+    options?: Partial<StallDetectionOptions>
+  ): StallDetectionOptions {
+    return {
+      enabled: options?.enabled ?? true,
+      stallTimeout: options?.stallTimeout ?? 30000,
+      checkInterval: options?.checkInterval ?? 5000,
+      minimumBytesPerSecond: options?.minimumBytesPerSecond ?? 1
+    }
+  }
+
+  /**
+   * Detect if current HttpStack supports progress events
+   */
+  private _supportsProgressEvents(): boolean {
+    const httpStack = this.options.httpStack
+    // Check if getName method exists and if it returns one of our known stacks
+    return typeof httpStack.getName === 'function' &&
+           ["NodeHttpStack", "XHRHttpStack"].includes(httpStack.getName())
```

Comment:
Instead of checking the name of the HTTP stack, it's better if the stack exposes its progress reporting capabilities with a newly added property or method.